core/Enclave/enclave_self_test.cpp

```diff
@@ -0,0 +1,939 @@
+                          VECTOR_LENGTH("tag"));
+
+    return (TEST_COMPARE(ciphertext) && TEST_COMPARE(tag));
+}
+
+static bool aes_gcm_decryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(aad);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+    GET_PARAMETER(tag);
+
+    uint8_t _plaintext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)aes_gcm_decrypt(&*key,
+                          _plaintext,
+                          get_block_mode(VECTOR_LENGTH("key")),
+                          &*ciphertext,
+                          VECTOR_LENGTH("ciphertext"),
+                          &*aad,
+                          VECTOR_LENGTH("aad"),
+                          &*iv,
+                          VECTOR_LENGTH("iv"),
+                          &*tag,
+                          VECTOR_LENGTH("tag"));
+
+    return TEST_COMPARE(plaintext);
+}
+
+static bool sm4_ctr_encryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _ciphertext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_ctr_encrypt(&*key,
+                          _ciphertext,
+                          &*plaintext,
+                          VECTOR_LENGTH("plaintext"),
+                          &*iv);
+
+    return TEST_COMPARE(ciphertext);
+}
+
+static bool sm4_ctr_decryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _plaintext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_ctr_decrypt(&*key,
+                          _plaintext,
+                          &*ciphertext,
+                          VECTOR_LENGTH("plaintext"),
+                          &*iv);
+
+    return TEST_COMPARE(plaintext);
+}
+
+static bool sm4_cbc_encryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _ciphertext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_cbc_encrypt(&*key,
+                          _ciphertext,
+                          &*plaintext,
+                          VECTOR_LENGTH("plaintext"),
+                          &*iv);
+
+    return TEST_COMPARE(ciphertext);
+}
+
+static bool sm4_cbc_decryption(map<string, string> test_vector)
+{
+    GET_PARAMETER(key);
+    GET_PARAMETER(plaintext);
+    GET_PARAMETER(iv);
+    GET_PARAMETER(ciphertext);
+
+    uint8_t _plaintext[VECTOR_LENGTH("plaintext")] = {0};
+    (void)sm4_cbc_decrypt(&*key,
+                          _plaintext,
+                          &*ciphertext,
+                          VECTOR_LENGTH("plaintext") + VECTOR_LENGTH("iv"),
+                          &*iv);
+
+    return TEST_COMPARE(plaintext);
+}
+
+static bool rsa_decryption(map<string, string> test_vector)
+{
```

Comment:
rsa_decryption -> rsa_decryption_with_xxx  (xxx is the mode you're testing)

Reply:
ok